### PR TITLE
SUBMARINE-1204. Notebook names cannot start with "submarine"

### PR DESCRIPTION
**Pull request by FYRichie**
### What is this PR for?
The name of the notebook cannot start with "submarine". Otherwise, the service name will conflict with the other services that the operator created.
For example, if the name of the notebook is "submarine-server", then it will conflict with the service "submarine-server" that the operator created.
This PR aims to avoid this problem by adding the notebookId in front of notebookName at k8sSubmitter, but the notebookId will not display on frontend.

### What type of PR is it?
[Bug Fix]

### What is the Jira issue?
https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-1204?filter=allopenissues
### How should this be tested?
Run the submarine service, and check `kubectl get notebook -n submarine-user-test`, if the notebook name in k8s is an <id>-<name> format, and the frontend only displays notebook name, then it is correct.
### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No

### CODE REVIEW

**Comment by KUAN-HSUN-LI** [comment]
@FYRichie Please fix the error. Thanks

File: submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java [hunk 1]
```diff
@@ -466,7 +466,9 @@ private boolean isDeploymentAvailable(String name) throws ApiException{
   @Override
   public Notebook createNotebook(NotebookSpec spec, String notebookId) throws SubmarineRuntimeException {
     Notebook notebook;
-    final String name = spec.getMeta().getName();
+    // final String name = spec.getMeta().getName();
```
Comment (pingsutw) [inline submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java:469]: nit: should we remove this

**Review by pingsutw — APPROVED** [review]

File: submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java (hunk 1 above)
Comment (FYRichie) [reply submarine-server/server-submitter/submitter-k8s/src/main/java/org/apache/submarine/server/submitter/k8s/K8sSubmitter.java:469]: @pingsutw I will remove this

**Comment by FYRichie** [comment]
@MortalHappiness @FYRichie Can you help me review my code?

**Review by MortalHappiness — APPROVED** [review]
LGTM

**Pull request closed without merging**